### ISSUE TITLE
Meta shortcode not resolved in DecoratedCodeBlock nodes

### ISSUE THREAD

**Issue by cderv** [quarto-dev/quarto-cli#6917]
See https://quarto.org/docs/output-formats/hugo.html#configuration

![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/ab87b239-b357-4046-9fc4-652ead268f02)

The `meta` is defined in main doc: https://github.com/quarto-dev/quarto-web/blob/34a49a835a864fb81f6f0a5370e9b417e418859e/docs/output-formats/hugo.qmd#L3
and used in included file inside a code block : https://github.com/quarto-dev/quarto-web/blob/34a49a835a864fb81f6f0a5370e9b417e418859e/docs/output-formats/_ssg-workflow.qmd#L62-L74

It works when used in main text: https://quarto.org/docs/output-formats/hugo.html#workflow
https://github.com/quarto-dev/quarto-web/blob/34a49a835a864fb81f6f0a5370e9b417e418859e/docs/output-formats/_ssg-workflow.qmd#L7

@cscheid I understood this was something we fixed and should work already. Is that so ?

**Comment by cderv** [comment]
Ok I got the reprex. This is an issue with adding `filename` attributes 

````markdown
---
title: "Test"
format: html
format-name: hugo
---

## Meta shortcode in text

This is format {{< meta format-name >}}

## Shortcode in code block

```yaml
format: {{< meta format-name >}}
```

## Shortcode in code block with filename attributes

```{.yaml filename=config.yml}
format: {{< meta format-name >}}
```
````

![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/5ab36e58-aa2c-4bb9-be3f-c0d6a79a187d)

**Comment by cderv** [comment]
@cscheid you may know more than me about this. 

I think the issue that Shortcodes is processed in `CodeBlock` but not in `DecoratedCodeBlock`. I could try solve it, but not sure of the best way to handle custom nodes here, as `DecoratedCodeBlock` should resolve to a `Codeblock` to be processed IMO.

**Comment by cscheid** [comment]
DecoratedCodeBlock has a CodeBlock inside it, so that CodeBlock should be getting processed. But this is clearly a bug. I'll investigate.

**Comment by cscheid** [comment]
This has been fixed by #6986 as well. On my `main`:

<img width="971" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/285675/e6fbb0a5-0389-4873-91d3-35914f2ba5ee">
